File: crates/filter-list-manager/src/manager/models/filter_list_rules_raw.rs
//! [`FilterListRules`]-counterpart represents `rules` and `disabled rules` as strings.
use crate::FilterId;

/// This contains `rules` and `disabled_rules` just as strings instead of vectors in the base type.
/// See the [`FilterListRules`] for more info.
#[derive(Clone)]
#[cfg_attr(test, derive(Debug))]
pub struct FilterListRulesRaw {
    /// Associated filter id.
    pub filter_id: FilterId,
    /// List of all rules in the filter list.
    pub rules: String,
    /// List of only disabled rules.
    pub disabled_rules: String,
    /// Rules count in this filter list. Simply a number of non-empty lines
    /// and does not start with a comment marker.
    pub rules_count: i32,
}

File: platform/windows/AdGuard.FilterListManager/MarshalLogic/FilterListRulesRaw.cs
namespace AdGuard.FilterListManager.MarshalLogic
{
    /// <summary>
    /// Raw version of <see cref="FilterListRules"/>
    /// </summary>
    public class FilterListRulesRaw
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="FilterListRulesRaw"/> class.
        /// </summary>
        /// <param name="filterId">The filter identifier.</param>
        /// <param name="rules">The rules.</param>
        /// <param name="disabledRules">The disabled rules.</param>
        /// <param name="rulesCount">The rules count.</param>
        public FilterListRulesRaw(int filterId, string rules, string disabledRules, int rulesCount)
        {
            FilterId = filterId;
            Rules = rules;
            DisabledRules = disabledRules;
            RulesCount = rulesCount;
        }
        
        /// <summary>
        /// Gets the filter identifier.
        /// </summary>
        public int FilterId { get; }
        
        /// <summary>
        /// Gets the rules of the filter.
        /// </summary>
        public string Rules { get; }

        /// <summary>
        /// Gets the disabled rules of the filter.
        /// </summary>
        public string DisabledRules { get; }

        /// <summary>
        /// Gets the rules count.
        /// </summary>
        public int RulesCount { get; }
    }
}
